### PR TITLE
Preparing next release 1.4.2 (third part)

### LINKED ISSUE #659: Google Play requeriment
App must target Android 14 (API level 34) or higher, deadline: `Aug, 30. 2024`

### PR DESCRIPTION
**Pull request by juanky201271**
- Android value transfers test fixed.
- Adding the concept of Value Transfers to History screen.
- Separate memos in the same array with a new line, like: `\n`.
- After `Send` or `Shield` the App scroll to top automatically in the history screen, the user needs to see the new/s transaction/s confirming.
- Adding an icon to help going to top of the list - history screen.
- Adding an icon to help going to top of the list - address book screen.
- #659 addressed.
- History line using the amount from Value Transfer, no fee added.
- If Tip to zingolabs is `enabled` and if you are sending to self -> the donation will send as well in this transaction.

### CODE REVIEW

File: app/utils/Utils.ts [hunk 1]
```diff
@@ -246,7 +242,7 @@ export default class Utils {
     // we need to exclude 2 use cases:
     // 1. send to self (make no sense to do a donation here)
```
Comment (Oscar-Pepper) [inline app/utils/Utils.ts:243]: i think this comment is outdated now?

File: components/History/History.tsx [hunk 2]
```diff
@@ -96,22 +170,23 @@ const History: React.FunctionComponent<HistoryProps> = ({
...
-          length={transactionsSorted.length}
-          tx={txDetail}
-          closeModal={() => setTxDetailModalShowing(false)}
-          openModal={() => setTxDetailModalShowing(true)}
+        visible={isValueTransferDetailModalShowing}
+        onRequestClose={() => setValueTransferDetailModalShowing(false)}>
+        <ValueTransferDetail
+          index={valueTransferDetailIndex}
+          length={valueTransfersSorted.length}
+          totalLength={valueTransfers.length}
```
Comment (Oscar-Pepper) [inline components/History/History.tsx:178]: i don't understand how the sorted length could be different to the value transfers length?

File: components/History/History.tsx [hunk 3]
```diff
@@ -45,42 +57,104 @@ const History: React.FunctionComponent<HistoryProps> = ({
...
+            // different txid
+            return txidComparison;
+          }
+        } else {
+          // different time
+          return timeComparison;
+        }
+      })
+      .slice(0, numVt);
+  }, [valueTransfers, numVt]);
```
Comment (Oscar-Pepper) [inline components/History/History.tsx:119]: i am not familiar with this code but does this only sort the value transfer or also combine them in some cases?

**Review by Oscar-Pepper — APPROVED** [review]
i have some comments but not blocking. GREAT JOB!

File: components/History/History.tsx (hunk 2 above)
Comment (juanky201271) [reply components/History/History.tsx:178]: in the fetch function you can see the `slice` with the first 50 lines... means the sorted list can have 50 and the total list can have N > 50...

File: components/History/History.tsx (hunk 3 above)
Comment (juanky201271) [reply components/History/History.tsx:119]: the `slice` cut the first `numVt` lines... the list in the beginning have the first 50 lines... and you can tap on `load more` until 100, and so on...

File: app/utils/Utils.ts (hunk 1 above)
Comment (juanky201271) [reply app/utils/Utils.ts:243]: upsss, yes. I'm going to fix in the next PR asap.

**Pull request merged**